.github/workflows/newman.yml

```diff
@@ -0,0 +1,147 @@
+      centreon_image:
+        required: true
+        type: string
+    secrets:
+      registry_username:
+        required: true
+      registry_password:
+        required: true
+
+jobs:
+  discover_collections:
+    runs-on: ubuntu-latest
+
+    outputs:
+      collections: ${{ steps.set_collections.outputs.collections }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: List Postman Collections and Environments
+        id: set_collections
+        run: |
+          collection_path="${{ inputs.collection_path }}"
+          collections=$(find "$collection_path" -type f -name "*.postman_collection.json" -printf "%P\n" | sort | jq -R -s -c 'split("\n")[:-1]')
+          echo "collections=$collections" >> $GITHUB_OUTPUT
+          echo "GITHUB_OUTPUT contents:"
+          cat $GITHUB_OUTPUT
+
+  postman_tests:
+   needs: discover_collections
+   runs-on: ubuntu-latest
+
+   strategy:
+    fail-fast: false
+    matrix:
+      feature: ${{ fromJson(needs.discover_collections.outputs.collections) }}
+
+   name: Postman Test - ${{ matrix.feature }}
+
+   defaults:
+    run:
+      shell: bash
+      working-directory: centreon/tests/rest_api
+   env:
+    CONTAINER_NAME: ${{ inputs.container_name }}
+    CENTREON_URL: ${{ inputs.centreon_url }}
+    CENTREON_IMAGE: ${{ inputs.centreon_image }}
+
+   steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+
+    - uses: pnpm/action-setup@v2
+      with:
+        version: 7
+
+    - name: Install Dependencies
+      run: pnpm install --frozen-lockfile
+      working-directory: centreon/tests/rest_api
+
+    - name: Login to registry
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}
+        username: ${{ secrets.registry_username }}
+        password: ${{ secrets.registry_password}}
+
+    - name: Start Centreon Web Container
+      run: docker run --name $CONTAINER_NAME -d -p 4000:80 --health-cmd="curl -f $CENTREON_URL/ || exit 1" --health-interval=5s $CENTREON_IMAGE
+
+    - name: Wait for Centreon Web Container to Start
+      run: |
+        pnpx wait-on http://localhost:4000/ --timeout 100000
+      timeout-minutes: 5
+
+    - name: Run Postman Tests and Generate HTML Report
+      run: |
+        collection_file="collections/${{ matrix.feature }}"
+        collection_name=$(basename "$collection_file" .postman_collection.json)
+        collection_directory="$(dirname "$collection_file")"
+        environment_file="$collection_directory/Environment-Collection.json"
+        if [ -f "$environment_file" ]; then
+          pnpm newman run "$collection_file" -e "$environment_file" --reporters cli,htmlextra,json-summary --reporter-htmlextra-title "$collection_name" --reporter-htmlextra-title "${collection_name// /_}" --reporter-htmlextra-export "newman/${collection_name}.html" --reporter-summary-json-export "postman_summaries/${collection_name}-summary.json" || true

```

Comment:
if you use `|| true`, this step will be always displayed in green state, even if an error happened

Reply:
it's ok, error is thrown after summary